### DISCUSSION TITLE
How do I toggle the lights in a room

### DISCUSSION THREAD

**Discussion by Fizedi** [cgiesche/streamdeck-homeassistant#283] [Help and Support] [answered]
Hi there! I'm stuck doing something I think should be easy (But often easy things in Home Assistant are not that easy...

I simply want a button that when I press it will turn off *all* the lights in the room. Press again and turn all them on. 

Thanks! (I hope it is a simple answer)

**Comment by cgiesche** [comment]
Hi @Fizedi,

it is very easy, but i will explain in some more details as the following often is a bit confuising:

Every button has two main configurations:
1. Keypad apperance
Here you can define what your button shows. It may be the state of a light or a temperature sensor. It is completely unrelated to what the button will do if you press it.

2. Keypad actions
This is where you tell the button what to do if you press it. In your case all you have to do is select "light" (or sometimes "switch") in the domain dropdown, then select "Toggle" in the Service dropdown and finally the light entity you want to toggle. Save at the configuration at the very bottom of the page. Done.

(Screenshot shows "turn on" service, but you can choose "toggle" to... toggle the light :))
![grafik](https://github.com/cgiesche/streamdeck-homeassistant/assets/5123180/80186326-14fc-483a-9c12-3ce1d9e36b5b)

**Reply by Fizedi** [reply]
Thanks for the quick reply! :)

You said:  and finally the light entity
As in singular. 

I'm looking to toggle all the lights in the room. In my excample in HA I have an Area called "Groom" but it is not visible in the Entity list. 

Thoughts?

**Comment by Fizedi** [comment, marked as answer]
I am seeing "Groups" in there. So I guess unless there is another way to add the area, I can makie a group. 

This worked... :)

Thanks for nudging me into the right direction.
